packages/backend/src/apps/gathersg/actions/update-case/index.ts

```diff
@@ -0,0 +1,188 @@
+          type: 'dropdown' as const,
+          showOptionValue: false,
+          required: true,
+          options: [
+            {
+              label: 'String',
+              value: ensureZodEnumValue(fieldTypeEnum, 'string'),
+            },
+            {
+              label: 'Number',
+              value: ensureZodEnumValue(fieldTypeEnum, 'number'),
+            },
+            {
+              label: 'Null',
+              value: ensureZodEnumValue(fieldTypeEnum, 'null'),
+            },
+          ],
+          customStyle: { flex: 1, minWidth: 0 },
+        },
+        {
+          placeholder: 'Value',
+          key: 'value',
+          type: 'string' as const,
+          required: true,
+          variables: true,
+          hiddenIf: {
+            fieldKey: 'fieldType',
+            op: 'equals',
+            fieldValue: 'null',
+          },
+          customStyle: { flex: 3, minWidth: 0, maxWidth: '60%' },
+        },
+      ],
+      hiddenIf: {
+        fieldKey: 'caseUuid',
+        op: 'is_empty',
+      },
+    },
+  ],
+
+  async run($) {
+    try {
+      const { caseUuid } = $.step.parameters
+      // Validation to prevent path traversals
+      validateDynamicFieldsAndThrowError({
+        caseUuid: String(caseUuid),
+      })
+
+      const payload = requestSchema.parse($.step.parameters)
+      const rawResponse = await $.http.patch('/cases/:caseUuid', payload, {
+        urlPathParams: {
+          caseUuid: $.step.parameters.caseUuid,
+        },
+      })
+      const response = responseSchema.parse(rawResponse.data)
+
+      $.setActionItem({
+        raw: {
+          ...response,
+        },
+      })
+    } catch (error) {
+      if (error instanceof ZodError) {
+        const firstError = fromZodError(error).details[0]
+        throw new StepError(
+          `${firstError.message}`,
+          GenericSolution.ReconfigureInvalidField,
+          $.step.position,
+          $.app.name,
+        )
+      }
+
+      if (error instanceof HttpError) {
+        // Case status is invalid
+        const { code, message, details } =
+          (error.response.data?.error as GatherSGError) || {}
+        if (error.response.status === 400 && code === 'RESOURCE_NOT_FOUND') {
+          throw new StepError(
+            message,
+            'Check that you have entered a valid case status.',
+            $.step.position,
+            $.app.name,
+          )
+        }
+
+        // Invalid field value type entered
+        if (error.response.status === 422 && code === 'INVALID_INPUT') {
+          const invalidFields = details?.fields as string[]
+          throw new StepError(
+            'Invalid field value type entered (between numbers, strings, etc)',
+            `Check that you have entered the correct value type for the following fields: ${invalidFields.join(
+              ', ',
+            )}`,
+            $.step.position,
+            $.app.name,
+          )
+        }
+      }
+
```

Comment:
can extract this out as a separate error handler to reuse across actions/dynamic data

Also handle 403 where api key does not have sufficient permissions
<img width="1128" height="191" alt="Image" src="https://github.com/user-attachments/assets/5813c394-6ae7-41c9-9fb9-79b110b9a953" />